src/integrations/fhir/clientInit.ts

```diff
@@ -11,28 +12,39 @@ import Client from 'fhirclient/lib/Client';
  * storage. If the auth token is not found in local storage, or is expired it will give a "not authorized" error. The
  * user must then reauthenticate by opening the window again from the EHR system to get a new launch url.
  *
- * @param reAuth set to true when launching a new context in a existing window/tab, to force a re-authentication
- * @param issuer
- * @param launch
+ * @param isLaunch set to true when launching a new context in a existing window/tab, to force a re-authentication
  */
-export const clientInitInBrowser = async (reAuth: boolean, issuer: string | undefined, launch: string | undefined): Promise<FhirService> => {
-    if (reAuth) {
+export const clientInitInBrowser = async (isLaunch: boolean): Promise<FhirApi> => {
+    if (isLaunch) {
         sessionStorage.clear();
     }
 
     const clientId: string = await fhirClientId();
     const client: Client = await oauth2.init({
         clientId: clientId,
-        iss: issuer,
-        launch: launch,
         scope: "launch patient/*.read openid fhirUser profile",
         redirectUri: "/"
     });
 
-    return new FhirService(client)
+    const clientInitIsComplete = client.patient.id !== null && client.state.tokenResponse?.access_token !== undefined;
+    if(!clientInitIsComplete) {
+        // TODO This should not happen according to how i interpret the client doc/code. Looks like some sort of race condition?
```

Comment:
Burde evt. legge til en feilhåndtering her?

Reply:
Det blir uansett kasta ein feil i dette tilfellet lenger ute, men den feilen blir på eit vis automatisk korrigert av klienten. Så ein opplever at når ein launche ny sesjon, så viser det ein feil i GUI ei lita stund, og så skjer det ei relasting av sida og då feiler det ikkje. 

Denne kodesnutten er igrunn berre midlertidig for å debugge og tydleggjere kva/kvifor det feiler. Eg er usikker på kvifor det skjer, og om det vil vere eit problem til slutt, men tenkte det var greit å ha denne koden for å hjelpe med å feilsøke meir i framtida. Trur derfor ikkje det vil vere ein fordel å kaste ein eigen feil her, men vi må nok sjå meir på feilen som oppstår seinare.